test/Infrastructure.IntegrationTest/AdminConsole/Migrations/FinalFlexibleCollectionsDataMigrationsTests.cs

```diff
@@ -0,0 +1,253 @@
+        Assert.Equal(OrganizationUserType.User, migratedOrgUser.Type);
+        Assert.Null(migratedOrgUser.Permissions);
+    }
+
+    [DatabaseTheory, DatabaseData(MigrationName = _migrationName)]
+    public async Task RunMigration_WithEditAndDeleteAssignedCollections_WithCustomUserType_MigratesToUserNullPermissions(
+        IUserRepository userRepository,
+        IOrganizationRepository organizationRepository,
+        IOrganizationUserRepository organizationUserRepository,
+        IMigrationTesterService migrationTester)
+    {
+        // Setup data
+        var orgUser = await SetupData(
+            userRepository, organizationRepository, organizationUserRepository,
+            OrganizationUserType.Custom, editAssignedCollections: true, deleteAssignedCollections: true);
+
+        // Run data migration
+        migrationTester.ApplyMigration();
+
+        // Assert that the user was migrated to a User type with null permissions
+        var migratedOrgUser = await organizationUserRepository.GetByIdAsync(orgUser.Id);
+        Assert.NotNull(migratedOrgUser);
+        Assert.Equal(orgUser.Id, migratedOrgUser.Id);
+        Assert.Equal(OrganizationUserType.User, migratedOrgUser.Type);
+        Assert.Null(migratedOrgUser.Permissions);
+    }
+
+    [DatabaseTheory, DatabaseData(MigrationName = _migrationName)]
+    public async Task RunMigration_WithoutAssignedCollectionsPermissions_WithCustomUserType_RemovesAssignedCollectionsPermissions(
+        IUserRepository userRepository,
+        IOrganizationRepository organizationRepository,
+        IOrganizationUserRepository organizationUserRepository,
+        IMigrationTesterService migrationTester)
+    {
+        // Setup data
+        var orgUser = await SetupData(
+            userRepository, organizationRepository, organizationUserRepository, OrganizationUserType.Custom,
+            editAssignedCollections: false, deleteAssignedCollections: false, accessEventLogs: true);
+
+        // Run data migration
+        migrationTester.ApplyMigration();
+
+        // Assert that the user kept the accessEventLogs permission and lost the editAssignedCollections and deleteAssignedCollections permissions
+        var migratedOrgUser = await organizationUserRepository.GetByIdAsync(orgUser.Id);
+        Assert.NotNull(migratedOrgUser);
+        Assert.Equal(orgUser.Id, migratedOrgUser.Id);
+        Assert.Equal(OrganizationUserType.Custom, migratedOrgUser.Type);
+        Assert.NotEqual(orgUser.Permissions, migratedOrgUser.Permissions);
+        Assert.NotNull(migratedOrgUser.Permissions);
+        Assert.Contains("accessEventLogs", orgUser.Permissions);
+        Assert.Contains("editAssignedCollections", orgUser.Permissions);
+        Assert.Contains("deleteAssignedCollections", orgUser.Permissions);
+
+        Assert.Contains("accessEventLogs", migratedOrgUser.Permissions);
+        Assert.True(migratedOrgUser.GetPermissions().AccessEventLogs);
```

Comment:
Can you take care of both of these `Dereference of a possibly null reference.` warnings? I'm fine with that being just `migratedOrgUser.GetPermissions()!.AccessEventLogs` if you are sure `GetPermissions` won't return null in this context. Or the longer way would be do do something like:
```suggestion
        var migratedOrgUserPermissions = migratedOrgUser.GetPermissions();
        Assert.NotNull(migratedOrgUserPermissions);
        Assert.True(migratedOrgUserPermissions.AccessEventLogs);
```